src/main/java/htsjdk/variant/vcf/VCFValidationFailure.java

```diff
@@ -0,0 +1,63 @@
+package htsjdk.variant.vcf;
+
+import htsjdk.utils.ValidationUtils;
+
+/**
+ * A class representing a VCF validation failure.
+ * @param <T> a type representing the object that is being validated
+ */
+class VCFValidationFailure<T> {
```

Comment:
This is returned by the public method `validateForVersion` so probably it should be public.  Or maybe the validate method should be protected.